signer/local_cosigner.go

```diff
@@ -280,14 +294,21 @@ func (cosigner *LocalCosigner) sign(req CosignerSignRequest) (CosignerSignRespon
 		return res, err
 	}
 
-	err = ccs.lastSignState.Save(SignStateConsensus{
+	// Create sign state with consensus lock
+	signStateConsensus := SignStateConsensus{
 		Height:                 hrst.Height,
 		Round:                  hrst.Round,
 		Step:                   hrst.Step,
 		Signature:              sig,
 		SignBytes:              req.SignBytes,
 		VoteExtensionSignature: res.VoteExtensionSignature,
-	}, &cosigner.pendingDiskWG)
+		ConsensusLock:          ccs.lastSignState.ConsensusLock, // Preserve existing lock
+	}
+
+	// Handle consensus lock updates according to Tendermint rules
+	signStateConsensus.ConsensusLock = updateConsensusLock(ccs.lastSignState.ConsensusLock, hrst.HRSKey(), req.SignBytes)
```

Comment:
the "update" prefix is implying that updateConsensusLock is mutating its argument(s). Consider something like "nextConsensusLock" or some other name which would imply that it is a pure function.